keras/applications/regnet.py

```diff
@@ -0,0 +1,1549 @@
+      into, only to be specified if `include_top` is True, and
+      if no `weights` argument is specified.
+    classifier_activation: A `str` or callable. The activation function to use
+      on the "top" layer. Ignored unless `include_top=True`. Set
+      `classifier_activation=None` to return the logits of the "top" layer.        
+  
+  Returns:
+  A `keras.Model` instance.
+  Raises:
+      ValueError: in case of invalid argument for `weights`,
+        or invalid input shape.
+      ValueError: if `classifier_activation` is not `softmax` or `None` when
+        using a pretrained top layer.
+      ValueError: if `include_top` is True but `num_classes` is not 1000.
+      ValueError: if `block_type` is not one of `{"X", "Y", "Z"}`
+  
+  """
+  if not (weights in {"imagenet", None} or tf.io.gfile.exists(weights)):
+    raise ValueError("The `weights` argument should be either "
+                     "`None` (random initialization), `imagenet` "
+                     "(pre-training on ImageNet), "
+                     "or the path to the weights file to be loaded.")
+
+  if weights == "imagenet" and include_top and classes != 1000:
+    raise ValueError("If using `weights` as `'imagenet'` with `include_top`"
+                     " as true, `classes` should be 1000")
+
+  # Determine proper input shape
+  input_shape = imagenet_utils.obtain_input_shape(
+      input_shape,
+      default_size=default_size,
+      min_size=32,
+      data_format=backend.image_data_format(),
+      require_flatten=include_top,
+      weights=weights)
+
+  if input_tensor is None:
+    img_input = layers.Input(shape=input_shape)
+  else:
+    if not backend.is_keras_tensor(input_tensor):
+      img_input = layers.Input(tensor=input_tensor, shape=input_shape)
+    else:
+      img_input = input_tensor
+
+  if input_tensor is not None:
+    inputs = layer_utils.get_source_inputs(input_tensor)
+  else:
+    inputs = img_input
+
+  x = inputs
+  if include_preprocessing:
+    x = PreStem(name=model_name)(x)
+  x = Stem(name=model_name)(x)
+
+  in_channels = 32  # Output from Stem
+
+  for num_stage in range(4):
+    depth = depths[num_stage]
+    out_channels = widths[num_stage]
+
+    x = Stage(block_type,
+              depth,
+              group_width,
+              in_channels,
+              out_channels,
+              name=model_name + "_Stage_" + str(num_stage))(x)
+    in_channels = out_channels
+
+  if include_top:
+    x = Head(num_classes=classes)(x)
+    imagenet_utils.validate_activation(classifier_activation, weights)
+
+  else:
+    if pooling == "avg":
+      x = layers.GlobalAveragePooling2D()(x)
+    elif pooling == "max":
+      x = layers.GlobalMaxPooling2D()(x)
+
+  model = training.Model(inputs=inputs, outputs=x, name=model_name)
+
+  # Load weights.
+  if weights == "imagenet":
+    if include_top:
+      file_suffix = ".h5"
+      file_hash = WEIGHTS_HASHES[model_name[-2:]][0]
```

Comment:
It looks like this part has not been tested? You are grabbing the last two characters, but your application keys above are all 4 characters long.

Reply:
Yes, actually. They are not tested yet. I'm still training the models.